src/ripple/core/tests/Coroutine.test.cpp

```diff
@@ -0,0 +1,109 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/test/jtx.h>
+#include <ripple/core/JobCoro.h>
+#include <chrono>
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+
+namespace ripple {
+namespace test {
+
+class Coroutine_test : public beast::unit_test::suite
+{
+public:
+    void
+    test_coroutine()
+    {
+        using namespace jtx;
+        Env env(*this);
+        std::atomic<int> i{0};
+        std::condition_variable cv;
+        auto& jq = env.app().getJobQueue();
+        jq.setThreadCount(0, true);
+        jq.postCoro(jtCLIENT, "Coroutine-Test",
+            [&](std::shared_ptr<JobCoro> jc)
+            {
+                std::thread t(
+                    [&i, jc]()
+                    {
+                        std::this_thread::sleep_for(
+                            std::chrono::milliseconds(10));
+                        ++i;
+                        jc->resume();
+                    });
+                jc->yield();
+                t.join();
+                ++i;
+                cv.notify_one();
+            });
+
+        {
+            std::mutex m;
+            std::unique_lock<std::mutex> lk(m);
+            cv.wait_for(lk, std::chrono::milliseconds(1000),
+                [&]() {return i == 2;});
```

Comment:
@miguelportilla Actually, I'm surprised by how unconfused I was about how the test works. My comment about the breakpoints because I _did_ put breakpoints on every line to watch it run, and I stepped slow enough that the `wait_for` timed out and everything aborted. Obviously, upping the timeout value gave me enough time.

BTW, now that I've seen the suggestion in one of my PRs, I realize `std::chrono::milliseconds(1000)` can be changed to `1000ms` if you bring `using namespace std::chrono;` into scope somewhere.